src/main/audio/server/ExternalAudioServer.cpp

```diff
@@ -14,7 +14,7 @@ void ExternalAudioServer::start()
 {
 	if (running)
 		return;
-	
+
```

Comment:
Are these formatting changes intentional? The ones where the spaces are removed are not a problem, as this project doesn't enforce formatting much. But there's another one that would be nice to revert.

Reply:
No, they are not.
I cannot remember making that change either.
I will rectify this in a moment.

Reply:
Are these changes found in this specific file?

Reply:
I believe it is fixed now.

Reply:
There are still changes to `ExternalAudioServer.cpp`. Did you push your changes? For some reason they are not included in this pull request.

Reply:
There - I've figured it out. VS puts some automatic formatting on the file. I had to modify them with Notepad++.

Reply:
The net changes are looking perfect 👍  

However, if you look at the commit history, you see there are 3 commits. If we merge your branch as is, this commit noise will become part of `ctoot`'s Git history. 2 of those 3 commits do not add much value to the Git history.

There are quite a few ways to avoid this completely, and there are quite a few ways to fix this from where you are now. I'll describe 2 ways to fix it from where you are now:

1. [Proper solution] In your local repo do
```
$ git reset --soft HEAD~2     # I'm assuming here that you haven't added any commits to this branch in the meantime
$ git add . && git commit -m "Expose PA-compatible audio server work method" && git push -f
```
That will rewrite the history and combine all changes in the last 3 commits of this branch into a new commit.

2. [Hacky solution but with perfectly fine end result] Start a new branch and manually apply the few changes you need. Open a new PR for this branch and close this one.

To avoid manual work when reverting files in the future, you can use the `git checkout -- ...` command I described in our Discord chat to revert a given file to its state at an arbitrary Git reference. In the example on Discord I've used a branch name, but it can be a commit hash or any other Git reference, such as `HEAD~1` if you know the desired untouched state resides in the previous commit.

If you haven't yet committed your changes, you can use `git restore ./path/to/file/or/dir`.

Finally, for future reference, to avoid commit noise you can keep adding modifications to your last commit by using `git commit --amend`. See for example https://www.atlassian.com/git/tutorials/rewriting-history

`amend` is a great way to reshape your work. I reach for it on a daily basis whenever I'm not interested in retaining every single change cycle in the VCS history. You get the advantages of having a safe anchor point you can always return to, without the disadvantages of accumulating uninteresting commits.

Reply:
Nice! Thanks for the tip about `amend`.

Reply:
Ah crap, I didn't give you the correct commands 😅 As you can see there are still 2 commits. Could you do the following to "squash" the 2 commits together into 1:
```
$ git reset --soft HEAD~1     # I'm assuming here that you haven't added any commits to this branch in the meantime
$ git add . && git commit --amend -m "Expose PA-compatible audio server work method" && git push -f
```

Reply:
I thought something was off about that. It's done now,